src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,227 @@
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+  const isTokenBridge = !isEthBridge
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+  const erc20L1Address = fromAssetId((isDeposit ? sellAsset : buyAsset).assetId).assetReference
+  const erc20L2Address = fromAssetId((isDeposit ? buyAsset : sellAsset).assetId).assetReference
+  const l1Provider = getEthersV5Provider(KnownChainIds.EthereumMainnet)
+  const l2Provider = getEthersV5Provider(KnownChainIds.ArbitrumMainnet)
+
+  if (isTokenBridge) {
+    // Since our related assets list isn't exhaustive and won't cut it to determine the L1 <-> L2 mapping, we double check that the bridge is valid
+    // by checking against Arbitrum bridge's own mappings, which uses different sources (Coingecko, Gemini, Uni and its own lists at the time of writing)
+    const arbitrumBridgeErc20L2Address = await (bridger as Erc20Bridger).getL2ERC20Address(
+      erc20L1Address,
+      l1Provider,
+    )
+    const arbitrumBridgeErc20L1Address = await (bridger as Erc20Bridger).getL1ERC20Address(
+      erc20L2Address,
+      l2Provider,
+    )
+
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L1Address), getAddress(erc20L1Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L1 ERC20 address: ${erc20L1Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L2Address), getAddress(erc20L2Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L2 ERC20 address: ${erc20L2Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+  }
+
+  if (sellAsset.assetId === usdcAssetId || sellAsset.assetId === usdcOnArbitrumAssetId) {
+    // https://www.circle.com/en/cross-chain-transfer-protocol
+    throw new Error('cctp not implemented')
+  }
+
+  if (isDeposit && !isEthBridge) {
+    const estimatedParentChainGas = fetchTokenFallbackGasEstimates()
+    const allowanceContract = await (bridger as Erc20Bridger).getL1GatewayAddress(
+      erc20L1Address,
+      l1Provider,
+    )
+
+    const { fast } = await adapter.getGasFeeData()
+
+    return Ok({
+      id: uuid(),
+      receiveAddress,
+      affiliateBps,
+      potentialAffiliateBps,
+      rate: '1',
+      slippageTolerancePercentageDecimal:
+        input.slippageTolerancePercentageDecimal ??
+        getDefaultSlippageDecimalPercentageForSwapper(SwapperName.ArbitrumBridge),
+      steps: [
+        {
+          // https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsSteps.tsx#L42
+          // 15 minutes in ms
+          estimatedExecutionTimeMs: 15 * 60 * 1000,
```

Comment:
Seems to consistently be less (around 10 minutes), but that's the same duration Arb bridge is using so